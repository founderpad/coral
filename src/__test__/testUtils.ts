// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"
import { render } from '@testing-library/react';
import { JSXElementConstructor, ReactElement } from 'react';

const Providers = ({ children }: { children: any }) => {
	return children;
	// return (
	//   <ThemeProvider theme="light">
	//     <TranslationProvider messages={defaultStrings}>
	//       {children}
	//     </TranslationProvider>
	//   </ThemeProvider>
	// )
};

const customRender = (
	ui: ReactElement<any, string | JSXElementConstructor<any>>,
	options = {}
): any => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
