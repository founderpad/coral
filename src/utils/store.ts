import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { persistReducer } from 'redux-persist';
import { PersistConfig } from 'redux-persist/es/types';
import storage from 'redux-persist/lib/storage';
import rootReducer from './reducer';

/**
 * Create the Redux store
 *
 * @author jlee
 */

// const isProd = process.env.NODE_ENV !== 'development';

/**
 * Persist storage configuration set up
 * @key[root]
 * @storage[storage]
 */
const persistConfig: PersistConfig<any> = {
	key: 'root',
	storage
};

const middleware = [...getDefaultMiddleware({ serializableCheck: false })];

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
	reducer: persistedReducer,
	middleware
	// devTools: isProd ?? false
});

// export { StoreProvider }

export default store;
