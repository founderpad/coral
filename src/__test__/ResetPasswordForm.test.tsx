import ResetPasswordFormForm from '@pages/resetpassword/components/ResetPasswordForm';
import userEvent from '@testing-library/user-event';
import store from '@utils/store';
import { Provider } from 'react-redux';
import { act, cleanup, fireEvent, render, waitFor } from './testUtils';

const setup = () => {
	const resetPasswordSetup = render(
		<Provider store={store}>
			<ResetPasswordFormForm />
		</Provider>
	);

	const resetPasswordForm = resetPasswordSetup.getByRole('form', {
		name: /resetpasswordform/i
	});
	const emailField = resetPasswordSetup.getByRole('textbox', {
		name: /email/i
	});
	const submitButton = resetPasswordSetup.getByRole('button', {
		name: /submit/i
	});

	return {
		resetPasswordSetup,
		resetPasswordForm,
		emailField,
		submitButton,
		...resetPasswordSetup
	};
};

const mockResetPassword = jest.fn();

jest.mock('@hooks/auth', () => ({
	useResetPassword: (): any => mockResetPassword
}));

jest.mock('next/router', () => ({
	useRouter() {
		return {
			route: '/resetpassword',
			query: 'rp_success'
		};
	}
}));

// jest.spyOn(require('next/router'), 'useRouter');

describe('Reset password form', () => {
	afterEach(cleanup);
	it('matches snapshot', async () => {
		const { asFragment } = setup();

		await waitFor(() => asFragment());
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render ResetPasswordForm', async () => {
		const { resetPasswordForm } = setup();
		expect(resetPasswordForm).toBeInTheDocument();
	});

	it('should render ResetPasswordForm fields', async () => {
		const { submitButton } = setup();
		expect(submitButton).toBeDisabled();
	});

	it('should make reset password request on submit', async () => {
		const { emailField, submitButton } = setup();

		expect(emailField).toBeInTheDocument();

		userEvent.type(emailField, 'jamie@gmail.com');
		await waitFor(() => expect(emailField).toHaveValue('jamie@gmail.com'));

		await act(async () => {
			fireEvent.click(submitButton);
		});

		expect(mockResetPassword).toHaveBeenCalledTimes(1);
	});

	it('should error on empty email on blur', async () => {
		const { resetPasswordSetup, emailField } = setup();

		expect(emailField).toBeInTheDocument();

		fireEvent.focus(emailField);
		userEvent.type(emailField, 'j@gmail');
		await waitFor(() => expect(emailField).toHaveValue('j@gmail'));

		fireEvent.blur(emailField);

		resetPasswordSetup.getByText('Please enter a valid email');
	});
});
