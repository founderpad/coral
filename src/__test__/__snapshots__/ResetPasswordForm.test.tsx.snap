// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset password form matches snapshot 1`] = `
<DocumentFragment>
  <form
    aria-label="reset-password-form"
    autocomplete="off"
    id="reset-password-form"
    name="reset-password-form"
    novalidate=""
  >
    <div
      class="chakra-stack css-w4pe59"
    >
      <p
        class="chakra-text css-yytlon"
      >
        Please enter your email address below and we will send you an email with instructions to reset your password.
      </p>
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <div
          aria-live="polite"
          class="css-jhkd4z"
          id="form-input"
        >
          <div
            class="css-qnoc8j"
          />
          <input
            aria-label="email"
            aria-required="true"
            class="chakra-input css-133xfak"
            errors="[object Object]"
            id="email"
            placeholder="Email"
            required=""
            value=""
          />
        </div>
      </div>
      <button
        aria-label="submit-reset-password"
        class="chakra-button css-1ktoofj"
        id="submit-reset-password"
        type="submit"
      >
        Reset password
      </button>
    </div>
  </form>
  <div
    class="css-18voih5"
  >
    <p
      class="chakra-text css-1u7an2r"
    >
      No account?
      <span
        class="chakra-link css-ntv2za"
        href="/register"
        title="Link to register an account"
      >
         Register now
      </span>
    </p>
    <span
      class="chakra-link css-1nbxiji"
      href="/login"
      title="Link to register an account"
    >
       Go to login
    </span>
  </div>
</DocumentFragment>
`;
